src/c++/perf_analyzer/mock_data_loader.h

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2022, NVIDIA CORPORATION. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include "data_loader.h"
+
+namespace triton { namespace perfanalyzer {
+
+class MockDataLoader : public DataLoader {
+ public:
+  MockDataLoader() = default;
+
+  cb::Error ReadDataFromStr(
```

Comment:
This function does _a lot_ of stuff which is unideal because it's largely duplicated code.

Can we put all the duplicated (from `ReadDataFromJSON()`) code from here into a new method in DataLoader so that all `ReadDataFromStr()` and `ReadDataFromJSON()` do is just read either a string or file into a `rapidjson::Document` and then call the refactored function that does all the validation on that document?

Reply:
discussed offline. We are going to do more refactoring in later tickets.

Reply:
I agree with Matt's statement. This doesn't seem like a mock to me.  This seems like a full dataloader.

If we want to use a mock, then these functions should do almost nothing other than counting how many times they were called. If we want to use dataloader functionality, then why not use the real dataloader?